### ISSUE TITLE
Re-export using `__all__`

### ISSUE THREAD

**Issue by JeppeKlitgaard** [google/jax#10062]
Currently the public JAX API often imports the actual implementations from `jax._src` (for example: [jax.random](https://github.com/google/jax/blob/a68b0f3a0a99d319fbfd7ddb7ff6942bf46f667e/jax/_src/random.py)).

IDE tools (pylance and others) and some linters (pylint and others) usually require re-exports to be done explicitly with an `__all__` tuple to prevent cluttering the namespace.

Using `__all__` should allow for better intellisense with `jax` for most modern editors. I believe `mypy` supports checking for this with the `--no-implicit-reexport` flag.

**Comment by JeppeKlitgaard** [comment]
Strangely I couldn't get this couldn't get my VSCode intellisense to work with JAX – it works now after restarting the editor. I think might have managed to have killed my pylance instance somehow. 

TL;DR: This absolutely works for me, apologies for the inconvenience!
